lib/logstash/inputs/unix.rb

```diff
@@ -140,23 +138,26 @@ def run(output_queue)
             raise
           end
         end
-      end # loop
+      end
     else
-      loop do
-        client_socket = UNIXSocket.new(@path)
-        client_socket.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
+      while !stop?
+        @client_socket = UNIXSocket.new(@path)
+        @client_socket.instance_eval { class << self; include ::LogStash::Util::SocketPeer end }
         @logger.debug("Opened connection", :client => @path)
-        handle_socket(client_socket, output_queue)
-      end # loop
+        handle_socket(@client_socket, output_queue)
+      end
     end
   end # def run
 
   public
-  def teardown
+  def stop
+    super
     if server?
       File.unlink(@path)
```

Comment:
I am a bit confused here, I know the code wasn't changed in this PR.
But `@path` is really used with the Unixsocket, which should be when `server? == false`, No?